optimum/amd/brevitas/configuration.py

```diff
@@ -0,0 +1,105 @@
+        apply_weight_equalization (`bool`, defaults to `False`):
+            Applies weight equalization accross layers, following https://arxiv.org/abs/1906.04721. This parameter is useful for models whose activation function is linear or piecewise-linear (like ReLU, used in OPT model), and allows to reduce the quantization error of the weights by balancing scales accross layers.
+        apply_gptq (`bool`, defaults to `False`):
+            Whether to apply GPTQ algorithm for quantizing the weights.
+        gptq_act_oder (`Optional[bool]`, defaults to `None`):
+            Whether to use activations reordering (act-order, also known as desc-act) when `apply_gptq=True`. If `apply_gptq=True`, defaults to `False`.
+    """
+
+    replace_mha_with_quantizable: bool = False
+    weights_bitwidth: int = 8
+    activations_bitwidth: int = 8
+    weights_param_method: Literal["stats", "mse"] = "stats"
+    weights_symmetric: bool = True
+    scale_precision: Literal["float_scale", "power_of_two_scale"] = "float_scale"
+    weights_quant_granularity: Literal["per_tensor", "per_channel", "per_group"] = "per_tensor"
+    weights_group_size: Optional[int] = None
+    quantize_zero_point: bool = True
+    activations_param_method: Literal["stats", "mse"] = "stats"
+    is_static: bool = False
+    activations_symmetric: bool = False
+    activations_quant_granularity: Literal["per_tensor", "per_row", "per_group"] = "per_tensor"
+    activations_group_size: Optional[int] = None
+    activations_equalization: Literal[None, "layerwise", "cross_layer"] = "cross_layer"
+    apply_weight_equalization: bool = False
+    apply_gptq: bool = False
+    gptq_act_oder: Optional[bool] = None
+
+    def __post_init__(self):
+        if self.activations_quant_granularity == "per_group" and self.activations_group_size is None:
+            self.activations_group_size = 64
+
+        if self.weights_quant_granularity == "per_group" and self.weights_group_size is None:
+            self.weights_group_size = 128
+
+        if self.is_static and self.activations_quant_granularity == "per_row":
+            raise ValueError('Static quantization and activations_quant_granularity="per_row" are incompatible.')
+
+        if self.apply_gptq and self.gptq_act_oder is None:
+            self.gptq_act_oder = False
+
+        if self.is_static and self.activations_quant_granularity != "per_tensor":
+            raise ValueError(
+                f'Static quantization with activations_quant_granularity="{self.activations_quant_granularity}" is not supported. The quantization granularity must be activations_quant_granularity="per_tensor" when using static quantization.'
+            )
+
+        if self.activations_quant_granularity == "per_row" and not self.replace_mha_with_quantizable:
+            raise ValueError("Per-row activations quantization requires setting replace_mha_with_quantizable to True.")
```

Comment:
Is this correct? To me `replace_mha_with_quantizable` has more to do with quantizing the `query @ key` and `softmax(state) @ value` as in https://github.com/Xilinx/brevitas/blob/925545b0d2d8ebc58544bbd2c6e3be8bb86a5b65/src/brevitas/nn/quant_mha.py#L546-L553 and https://github.com/Xilinx/brevitas/blob/925545b0d2d8ebc58544bbd2c6e3be8bb86a5b65/src/brevitas/nn/quant_mha.py#L563-L566, is it not?

Reply:
There was a limitation that would prevent us having per_row without replacing.
This will be fixed [with this PR](https://github.com/Xilinx/brevitas/pull/818).

Reply:
I see. Should the error be about `query @ key` & `softmax(state) @ value` quantization, and have this in the config?

Reply:
What would go in the config?

Reply:
That we attempt to quantize `query @ key` and `softmax(state) @ value` (https://github.com/Xilinx/brevitas/blob/925545b0d2d8ebc58544bbd2c6e3be8bb86a5b65/src/brevitas/nn/quant_mha.py#L563-L566) vs not doing so